### PR TITLE
Streamlining the login or homescreen VC presentation.

### PR DESCRIPTION
**Pull request by peytonshetler**
## Summary

I wasn't happy with the way that the app would show the homescreen for a split second before it would decide whether or not to kick the user out. 

What I did:

1. Added a Launchscreen.storyboard just for the heck of it really

2. Added a splash screen that contains most of the logic to check if there's a user.  It basically looks just like the launch screen so that it doesn't LOOK like it's presenting too many things before login.

Only mark Feature completed if you need the Card pulled to done when this PR is approved and merged into master

#### Characteristics

- [X] Feature completed
- [X] Experimental
- [X] UI affected (screen shots included)

#### Testing

- [ ] included
- [X] Not Included

### CODE REVIEW

**Comment by peytonshetler** [comment]
Hmmm, I'd love to create a Helper class that contains all of the login/logout functions but I'm running into an issue where I can't use the present( ) method b/c the class isn't a view controller.  I could've sworn I've done something like this before but I can't remember how??

I'd love to be able to call `Helper.handleLogout( )` from any view controller lol.

**Comment by JZDesign** [comment]
Easy. Add `inPresentingContext context: UIViewController` to the method parameters. Then you could do `context.present(authStuff, animated: true)`

File: Been-There/Been-There/Controller/SplashScreen/UI+SplashScreenViewController.swift [hunk 1]
```diff
@@ -0,0 +1,34 @@
+//
+//  UI+SplashScreenViewController.swift
+//  Been-There
+//
+//  Created by Peyton Shetler on 10/28/19.
+//  Copyright © 2019 Peyton Shetler. All rights reserved.
+//
+import UIKit
+
+extension SpashScreenViewController {
```
Comment (JZDesign) [inline Been-There/Been-There/Controller/SplashScreen/UI+SplashScreenViewController.swift]: Even better if you extend `UIViewController`

File: Been-There/Been-There/Helper/Helper.swift [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+    //  TODO: NEED TO FIGURE OUT HOW TO MOVE THESE TO THIS FILE
+    
+//    func checkIfUserIsLoggedIn() {
+//        let userID = Auth.auth().currentUser?.uid
+//
+//        if userID != nil {
+//            proceedToHomescreen()
+//        } else {
+//            handleLogout()
+//        }
```
Comment (JZDesign) [inline Been-There/Been-There/Helper/Helper.swift]: I would do something like

```swift

struct AuthHandler {
    static func handleAuthenticationState(inPresentingContext context: UIViewController) {
        if isLoggedIn() {
            context.proceedToHome()
        } else {
            context.showLandingScreen
        }
    }
    
    static func isLoggedIn() -> Bool {
        return (Auth.auth().currentUser?.uid != nil) ? true : false
    }
}

extension UIViewController {
    func proceedToHome() {}
    func showLandingScreen {}
}
```

Then you can call it from any view controller like: `AuthHandler.handleAuthenticationState(inPresentingContext: self)`

**Comment by peytonshetler** [comment]
I changed a lot of my auth handler logic and, while it worked before, it's MUCH cleaner now haha.  Thanks!!

**Pull request merged**